DAOS-4587 test: Add dmg_pool helpers to libdaos_tests

mjmac opened the pull request:
These helpers use dmg to perform pool management functions, and replace the
following mgmt API methods:
  * daos_pool_create()
  * daos_pool_destroy()
  * daos_pool_list()

Additional improvements:
  * Add a retry on pool create for -DER_TIMEDOUT
  * Allow dmg callers to retrieve DaosStatus via JSON, when available

Test updates:
  * Disable secure mode for zero_config test for now, due to daos_racer
    running on a node without admin certificates

daosbuild1 commented on src/common/tests_dmg_helpers.c:
```diff
@@ -0,0 +1,573 @@
+	struct passwd		*passwd = NULL;
+	struct group		*group = NULL;
+	struct daos_prop_entry	*entry;
+	char 			tmp_buf[L_tmpnam], *tmp_name = tmp_buf;
```
(style)  please, no space before tabs

daosbuild1 replied:
```suggestion
	char			tmp_buf[L_tmpnam], *tmp_name = tmp_buf;
```

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/3/execution/node/61/log

daosbuild1 commented:
Test stage Build on Ubuntu 20.04 with Clang completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/3/execution/node/261/log

daosbuild1 commented:
Test stage Functional_Hardware_Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/4/execution/node/911/log

daosbuild1 commented:
Test stage Functional_Hardware_Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/5/execution/node/867/log

daosbuild1 commented:
Test stage Functional_Hardware_Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/6/execution/node/905/log

daosbuild1 commented:
Test stage Functional_Hardware_Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/9/execution/node/859/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/10/execution/node/254/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/10/execution/node/195/log

daosbuild1 commented:
Test stage Build on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/10/execution/node/258/log

daosbuild1 commented:
Test stage Build on Ubuntu 20.04 with Clang completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/10/execution/node/261/log

daosbuild1 commented:
Test stage Build on CentOS 7 debug completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/10/execution/node/293/log

daosbuild1 commented:
Test stage Build on CentOS 7 release completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/10/execution/node/292/log

daosbuild1 commented:
Test stage Build on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3262/10/execution/node/294/log

mjmac commented:
@wangdi1: I incorporated your fix for the service ranks, will add a co-authored-by line to the final commit message if that's OK. I also addressed the nvme_size issue by allowing the rebuild suite to set it to 0 so that those tests don't request an invalid NVME size.

mjmac commented:
This PR finally got a green run on all of the tests, but was marked fail due to unrelated memory leaks in cr_hg_init().

mjmac commented:
@amjustin13: Add you for review of the test modifications. Basically, we need to tell daos_test not to request NVMe for pools if the config doesn't include NVMe.
